dashboard/test/controllers/data_docs_controller_test.rb

```diff
@@ -26,12 +30,12 @@ class DataDocsControllerTest < ActionController::TestCase
   test_user_gets_response_for :create, params: -> {@test_params}, user: nil, response: :redirect
   test_user_gets_response_for :create, params: -> {@test_params}, user: :student, response: :forbidden
   test_user_gets_response_for :create, params: -> {@test_params}, user: :teacher, response: :forbidden
-  test_user_gets_response_for :create, params: -> {@test_params}, user: :levelbuilder, response: :success
+  test_user_gets_response_for :create, params: -> {{key: 'unique_key'}}, user: :levelbuilder, response: :found
```

Comment:
This was returning `success` because it was hitting an error that the key was not unique, but `render :not_acceptable, json: @data_doc.errors` renders a status code of `success`

Reply:
Nice catch! Out of curiosity, is the only reason the 3 preceding tests are not running into that same issue because it hits the login/forbidden issue first? If so, could it be misleading to use @test_params in those cases since if something changed in the future that would accidentally allow, say, a teacher to access it, the error from @test_params not having a unique key may still return a success as a false positive?

Reply:
Oh true, plus you have the comment above to specify to anyone reading the test that the change in permissions to levelbuilder is what changes the test's expected response (my previous worry was confusion over what was changing between tests since the key changed too). I think it's actually good the way you have it (or all with 'unique_key' if you think that's better) but probably no need for each to have it's own key.

Reply:
A few thoughts here on redirects:
1. please disambiguate from a redirect to sign in, by using `redirected_to:` (as in earlier comment)
2. optional style point: you could consider using `:redirect` instead of `:found` make this read a little easier. lmk if you need help finding the docs for what symbols you can pass in here.

Reply:
in terms of `@test_params`, it does seem a little weird to be testing what happens in the case where an existing key is specified. It seems like the most important thing to test is what happens when different users try to create an available key, and then have just one test verify what happens when the key already exists.

Reply:
Good call on all of these –– will update!